### DISCUSSION TITLE
Using AnimationPlayer::seek_to – is this a bug, or am I using it wrong?

### DISCUSSION THREAD

**Discussion by ian-h-chamberlain** [bevyengine/bevy#11159] [Q&A] [answered]
Hello, I was trying to debug an animation issue on a project of mine, initially thought it might be fixed by the 0.12.1 animation changes, but now I'm not sure. 

I tried a minimal reproduction using the `animated_transform` example with two cases, and I was surprised to see similar behavior to my project. Essentially, I am trying to `seek_to` the end of the clip, and I expected the result to look the same as the end of playing the animation, but it looks different!

<details>
<summary>First case (play animation to end)</summary>

```diff
diff --git a/examples/animation/animated_transform.rs b/examples/animation/animated_transform.rs
index 0a3aa0239..4246c659a 100644
--- a/examples/animation/animated_transform.rs
+++ b/examples/animation/animated_transform.rs
@@ -109,11 +109,11 @@ fn setup(
         },
     );
 
     // Create the animation player, and set it to repeat
     let mut player = AnimationPlayer::default();
-    player.play(animations.add(animation)).repeat();
+    player.play(animations.add(animation));
 
     // Create the scene that will be animated
     // First entity is the planet
     commands
         .spawn((
```
Screenshot after the animation has played:
<img width="1392" alt="0" src="https://github.com/bevyengine/bevy/assets/11131775/126489cf-a6cd-4de2-99d8-6bfbe17c01eb">

</details>

<details>
<summary>Second case (seek to end of clip)</summary>

Second case:
```diff
diff --git a/examples/animation/animated_transform.rs b/examples/animation/animated_transform.rs
index 0a3aa0239..f6e8b456e 100644
--- a/examples/animation/animated_transform.rs
+++ b/examples/animation/animated_transform.rs
@@ -109,11 +109,12 @@ fn setup(
         },
     );
 
     // Create the animation player, and set it to repeat
     let mut player = AnimationPlayer::default();
-    player.play(animations.add(animation)).repeat();
+    let duration = animation.duration();
+    player.play(animations.add(animation)).seek_to(duration);
 
     // Create the scene that will be animated
     // First entity is the planet
     commands
         .spawn((
```

Screenshot:
<img width="1392" alt="0" src="https://github.com/bevyengine/bevy/assets/11131775/ce706b50-51b7-434f-a0ff-e8723e8db5de">

</details>

Am I missing something about how this API is supposed to work, or is this something that could be considered a bug? If so, is there some way I can work around it so that the objects are spawned as if they had played their animations to completion? Thanks!

**Comment by A-Walrus** [comment, marked as answer]
I think this might be related to #10832

**Reply by ian-h-chamberlain** [reply]
Ahh, thank you! I was able to use the changes from #10842 to fix the issue I was seeing, both in the `animated_transform` example and my project! Hopefully that change makes it in for 0.13 but in the meantime I can use your fork, so thanks for the pointer.
